runtime/vm/vmthread.cpp

```diff
@@ -267,6 +273,11 @@ allocateVMThread(J9JavaVM * vm, omrthread_t osThread, UDATA privateFlags, void *
 	if (J9_XACCESS_NONE != vm->safePointState) {
 		setHaltFlag(newThread, J9_PUBLIC_FLAGS_HALTED_AT_SAFE_POINT);
 	}
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	if (VM_CRIUHelpers::isJVMInSingleThreadMode(vm) && VM_VMHelpers::threadCanRunJavaCode(newThread)) {
+		vm->internalVMFunctions->setHaltFlag(newThread, J9_PUBLIC_FLAGS_HALT_THREAD_FOR_CHECKPOINT);
```

Comment:
No need to go through the function table. In fact, the inline versions could be used now this is a cpp file, but let's leave that for another time.